pandas/_typing.py

```diff
@@ -343,6 +343,12 @@ def closed(self) -> bool:
 # dropna
 AnyAll = Literal["any", "all"]
 
+# merge
+MergeHow = Literal["left", "right", "inner", "outer", "cross"]
+
+# join
+JoinHow = Literal["left", "right", "inner", "outer"]
```

Comment:
Join supports cross as well

Reply:
typed join with `MergeHow`.. used `JoinHow` for the indices.

I actually took this from pandas-stubs

https://github.com/pandas-dev/pandas-stubs/blob/db3b883dbe8d22863c8aca37c89431afce3b57e2/pandas-stubs/_typing.pyi#L299-L300

Reply:
Could you check how they type DataFrame.join?

Reply:
Yep